lib/oplog/oplogEntry.go

```diff
@@ -148,6 +130,23 @@ func (op *oplogEntry) ChangedFields() []string {
 	return []string{}
 }
 
+// Given a bson.Raw object, returns the top level keys of the document it represents
+func mapKeysRaw(rawData bson.Raw) []string {
+	elements, err := rawData.Elements()
+	if err != nil {
+		log.Log.Errorw("Failed to unmarshal oplog data",
+			"error", err)
+		return []string{}
+	}
```

Comment:
If the original one had an empty dataset (most likely reason for failing here), it would also return the same empty value.  So I think it's accurately following existing behavior, but it's another case where refactoring to return errors might make sense.

Reply:
Is that right? The documentation implies that this means the bson is malformed somehow. I may be misunderstanding. https://pkg.go.dev/go.mongodb.org/mongo-driver/bson#Raw.Elements

Reply:
I did add an error that will bubble up to the level where the original unmarshal was